docs/how_it_works.md

```diff
@@ -0,0 +1,132 @@
+
+### Transformations
+
+The `StreetNetwork` is techncially usable at this point, but it's still very close to OSM -- which is both under-specified (lane width is almost never tagged, but we need to render something) and imprecise. The rest of the magic happens by calling `apply_transformations`. This performs the specified steps in order. `apply_transformations_stepwise_debugging` can be used by UIs to preserve the intermediate `StreetNetwork` after each step, for debugging and understanding the transformations.
+
+The caller explicitly lists the transformations they want, in order. `standard_for_clipped_areas` is a good list to start with. Splitting things into explicit steps like this is good:
+
+- Users can opt into experimental steps
+- Some callers may not want to deviate too far from OSM, because they're using osm2streets to edit directly
+- Incremental debugging
+
+But it's also confusing in a few ways:
+
+- Roads and intersections both contain derived state. When we modify something, we may need to re-run some transformations. For example, after collapsing sausage links, a road's trimmed road geometry changes, so we may need to detect and merge short roads again. Effects from one transformation may need to propagate to adjacent roads and intersections. The dataflow is implicit; the caller must deal with it manually.
+- Some transformations fill out state that's pretty fundamental, like intersection complexity and movements. This maybe shouldn't be expressed as a transformation and should happen more upfront, like how `Road::new` immediately determines lanes from OSM tags.
+
+## Operations on a StreetNetwork
+
+Some of the below transformations do something complicated worth calling out. Likely this code shouldn't be in `transform/`.
+
+### Collapsing an intersection
+
+`collapse_intersection` removes an intersection that has only two roads connected to it. Transformations below describe when this is called. The operation itself mostly just stitches together the geometry of the two roads in a straightforward way. It fixes turn restrictions (which should get simpler with opaque IDs). The caller is responsible for fixing up lanes between the two roads -- or rather, making sure they match up compatibly in the first place.
+
+### Merging a short road
+
+`merge_short_road` removes a road, then combines the two intersections into one.
+
+It first calculates trimmed intersection geometry for the two intersections. On each connected road (besides the short one being merged, of course), we store the trimming distance in `trim_roads_for_merging`, so that later the intersection geometry algorithm can follow a special case for the single merged intersection.
+
+Most of the complexity here comes from fixing up IDs and turn restrictions. Opaque IDs likely make this a much simpler operation!
+
+## The transformations
+
+TODO: before/after pictures
```

Comment:
I don't think it's worth getting so carried away as to add screenshots here. I think this is valuable as a jumping off point that should get shorter and shorter as the structure of the code gets clearer, and as the docstrings improve.

Reply:
Fair enough. At some point for explaining purposes, it'd be nice to have a "gallery" of all the transformations available. Maybe if we plumb more info into the street-explorer URLs, we could eventually express "run with debug steps, and show step 5"